tests/__init__.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+from typing import Callable
+
+
+def check(actual: object, klass: type, dtype: type | None = None) -> None:
+
+    if not isinstance(actual, klass):
+        raise RuntimeError(f"Expected type '{klass}' but got '{type(actual)}'")
+    if dtype is None:
+        return None
+
+    if hasattr(actual, "__iter__"):
+        value = next(iter(actual))  # type: ignore[call-overload]
+    else:
+        value = actual.left  # type: ignore[attr-defined]
```

Comment:
checking the actual value is imho better than checking `.dtype`:

- users care about the actual values and
- it simplified the type checking (otherwise we have sometimes a np.dtype object, sometimes an actual type, and sometimes a function)

Reply:
Maybe we should check value _and_ `.dtype`.  For things like `Series[bool]`, we want to make sure that the `dtype` correspods to the generic attribute.

Reply:
I will try it but I think that dtype is often a numpy dtype while the values (when retrieving them) are builtin types

Reply:
I prefer to revisit that maybe in the future: dtype is  often a numpy dtype which is not compatible with the type of the value (str vs. numpy.dtype[object_], ...).